assets/javascripts/discourse/routes/user-activity-portfolio.js.es6

```diff
@@ -0,0 +1,24 @@
+import UserTopicListRoute from "discourse/routes/user-topic-list";
+import UserAction from "discourse/models/user-action";
+
+export default UserTopicListRoute.extend({
+  userActionType: UserAction.TYPES.topics,
+
+  model: function() {
+
+    let filter = Discourse.SiteSettings.topic_list_activity_portfolio_filter;
+    let filter_delimiter = filter.indexOf(':');
+    let filter_type = filter.substring(0, filter_delimiter);
+    let filter_parameter = filter.substring(filter_delimiter + 1);
+
+    if (filter_type == "category") {
+      return this.store.findFiltered("topicList", {
+        filter: "topics/created-by/" + this.modelFor("user").get("username_lower"),
+        params: {category: filter_parameter}
+      })} else {
+        return this.store.findFiltered("topicList", {
+          filter: "topics/created-by/" + this.modelFor("user").get("username_lower"),
+          params: {tags: filter_parameter}

```

Comment:
@merefield I'm not across the spec, but should we also add a topicList query option here that filters the topic list to only include topics with an associated thumbnail? I suspect it's this way because filtering by category and tag works for Bart and P16, where all topics with a particular category or tag have thumbails, however there may be:

1. Instances in which that doesn't hold true for bart and p16. Would they still want non-image topics to appear here?

2. Instances for other users where this filter won't produce image-only results.

Considering they're the sponsors, perhaps we just need to create a new open source task for this update at this stage and work on it down the track.

Reply:
I guess the name 'Portfolio' was chosen later in the game.  It was initially called Favourites, but Bart & I preferred the former.  Yeah, I'd prefer if this was demand lead.  Tagging or Categorising can be used carefully to ensure this is never an issue.  The wording can also be customised to something else if people prefer I believe from the Discourse front end.